depenguinme.sh

```diff
@@ -179,7 +182,12 @@ fi
 distro=$(/usr/bin/lsb_release -i | awk '{print $3}')
 
 if [[ "$distro" == "Debian" ]]; then
-	DEPS+=("genisoimage:genisoimage")
+        DEPS+=("genisoimage:genisoimage")
+        # download hpe signing keys
+        curl http://downloads.linux.hpe.com/SDR/hpPublicKey1024.pub       | apt-key add -
+        curl http://downloads.linux.hpe.com/SDR/hpPublicKey2048.pub       | apt-key add -
+        curl http://downloads.linux.hpe.com/SDR/hpPublicKey2048_key1.pub  | apt-key add -
+        curl http://downloads.linux.hpe.com/SDR/hpePublicKey2048_key1.pub | apt-key add -
```

Comment:
A couple of issues:
- The script uses tabs (not spaces) - this is not a pure style decision, but there are also (small) functional differences in bash
- Fetching something important like signing keys should use https to prevent MITM attacks
- A 1024 bit key seems very weak in this day and age, I wouldn't bother adding that one
- This script uses wget everywhere else, so lets stick to using one tool for the job (we could also replace wget with curl of course, I think we use wget, as it was available with Hetzner when we started)
- Looking at https://downloads.linux.hpe.com/SDR/keys.html it seems like that there is only one key is really required for packages signed after 2015 (I can't imagine that any packages older than that are required?). They also don't list the 1024 bit key.

```
HPE Public Keys
It is suggested that you enroll all keys to verify current and older versions of packages hosted in SDR repositories:

      hpPublicKey2048.pub	for packages published during 2014	fingerprint: 5CE2D476
      hpPublicKey2048_key1.pub	for packages published during 2015	fingerprint: B1275EA3
      hpePublicKey2048_key1.pub	for packages published after 2015	fingerprint: 26C2B797
```

Checking the key fingerprint would be kind of nice, but maybe is a bit overkill (I'm not sure if we have GNUPG available).

Simple idea: What about just including the one key required for signing packages into the code, getting rid of the need (and risk) of fetching it every time?

```suggestion
	DEPS+=("genisoimage:genisoimage")
	# Add public key used for signing HPE packages since 2016, see
	# https://downloads.linux.hpe.com/SDR/keys.html
	cat <<-"EOH" | apt-key add -
	-----BEGIN PGP PUBLIC KEY BLOCK-----
	Version: GnuPG v1.4.12 (GNU/Linux)

	mQENBFZp0LkBCACXajRw3b4x7G7dulNYj0hUID4BtVFq/MjEb6PHckTxGxZDoQRX
	RK54tiTFA9wq3b4P3yEFnOjbjRoI0d7Ls67FADugFO+cDCtsV9yuDlaYP/U/h2nX
	N0R4AdYbsVd5yr6xr+GAy66Hmx5jFH3kbC+zJpOcI0tU9hcyU7gjbxu6KQ1ypI2Q
	VRKf8sRBJXgmkOlbYx35ZUMFcmVxrLJXvUuxmAVXgT9f5M3Z3rsGt/ab+/+1TFSb
	RsaqHsIPE0QH8ikqW4IeDQAo1T99pCdf7FWr45KFFTo7O4AZdLMWVgqeFHaSoZxJ
	307VIINsWiwQoPp0tfU5NOOOwB1Sv3x9QgFtABEBAAG0P0hld2xldHQgUGFja2Fy
	ZCBFbnRlcnByaXNlIENvbXBhbnkgUlNBLTIwNDgtMjUgPHNpZ25ocEBocGUuY29t
	PokBPQQTAQIAJwUCVmnQuQIbLwUJEswDAAYLCQgHAwIGFQgCCQoLAxYCAQIeAQIX
	gAAKCRDCCK3eJsK3l9G+B/0ekblsBeN+xHIJ28pvo2aGb2KtWBwbT1ugI+aIS17K
	UQyHZJUQH+ZeRLvosuoiQEdcGIqmOxi2hVhSCQAOV1LAonY16ACveA5DFAEBz1+a
	WQyx6sOLLEAVX1VqGlBXxh3XLEUWOhlAf1gZPNtHsmURTUy2h1Lv/Yoj8KLyuK2n
	DmrLOS3Ro+RqWocaJfvAgXKgt6Fq/ChDUHOnar7lGswzMsbE/yzLJ7He4y89ImK+
	2ktR5HhDuxqgCe9CWH6Q/1WGhUa0hZ3nbluq7maa+kPe2g7JcRzPH/nJuDCAOZ7U
	6mHE8j0kMQMYjgaYEx2wc02aQRmPyxhbDLjSbtjomXRr
	=voON
	-----END PGP PUBLIC KEY BLOCK-----
	EOH
```

Reply:
thats actually a great idea! Sorry about the formatting, I think my editor went rogue with that one.